src/jobs/jobs.controller.ts

```diff
@@ -335,23 +318,27 @@ export class JobsController {
   /**
    * Checking if user is allowed to create job according to auth field of job configuration
    */
+  // TBD
   async instanceAuthorization(
-    jobCreateDto: CreateJobDto,
+    jobParams: Record<string, any> | undefined,
```

Comment:
`instanceAuthorization` is currently implemented to accomodate both `create` and `update`.

What about get, delete etc? What would be the auth options for these? Would the same function be sufficient or should it be defined differently for each?

> TODO: Update needs an additional specific user that we set in configuration.
We might want to create a user with a specific jwt token dedicated for the specific job update.  I see this a update segregation. ---> @nitrosx 's comment

Reply:
The only other operation that we should allow is GET.
We should  brainstorm and come up with the permission matrix.

Reply:
Use params from config, not the DTO.

Reply:
`instanceAuthorization` is different for `create` vs `update`.
For `create`, after endpoint auth we need to check if the user has access to the datasets.
For `update`, after endpoint auth we need to check whether the user has the right to update this specific job. To be covered by @nitrosx in the endpoint authorization.